### PR TITLE
🐛 Ensure uploaded outputs always have a unique S3 object name

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

**NOTE:** requires @odeimaiz magic to make frontend work correctly (see Converter.js, FolderViewer.js)

say we have a node X in a project Y
BEFORE:
```bash
/outputs/output_1/file
/outputs/output_2/file
```
both would end up in S3 under 
```bash
BUCKET_NAME/PROJECT_Y/NODE_X/file
```
--> triggering all kind of issues (503s, maybe 404s)

AFTER:
will end up respectively in
```bash
BUCKET_NAME/PROJECT_Y/NODE_X/outputs/output_1/file
BUCKET_NAME/PROJECT_Y/NODE_X/outputs/output_2/file
```

image
image
image
image

~~This is not the ideal solution, since this means that a compression step is added even for a single file. BUT, this is the fastest fix.
The real fix will come once the storage micro-service can handle folders which is planned for later and then also the S3 object name shall become something like:~~
**IMPORTANT NOTE:** The previous simple fix did not work as expected due to issue when pulling the data back with ```file_to_key_map``` enabled (as shown by tests - which is why we need them). Therefore the more correct solution was implemented, which implies:

- changes in storage micro-service to allow simcore S3 file IDs to be changed from ```PROJECT_ID/NODE_ID/FILE.EXT``` to ```PROJECT_ID/NODE_ID/ANY/NUMBER/OF/SUBFOLDERS/FILE.EXT``` as S3 object name.
- this should be a pre-requisite to upload folders directly without zipping

Bonus:
- reduce number of progress bar updates when uploading

## Related issue/s
fixes https://github.com/ITISFoundation/osparc-issues/issues/752

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]

**Review by Surfict — APPROVED** [review]

File: packages/simcore-sdk/src/simcore_sdk/node_ports_common/data_items_utils.py [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+    node_id: str,
+    *,
+    file_base_path: Optional[Path] = None,
 ) -> SimcoreS3FileID:
-    return SimcoreS3FileID(f"{project_id}/{node_id}/{file_path.name}")
+    s3_file_name = file_path.name
+    if file_base_path:
+        s3_file_name = f"{file_base_path / file_path.name}"
+    clean_path = Path(f"{project_id}/{node_id}/{s3_file_name}")
+    return SimcoreS3FileID(f"{clean_path}")
```
Comment (GitHK) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_common/data_items_utils.py:20]: I see, this is the actual fix. 👍

**Review by GitHK — COMMENTED** [review]
I have one concern. Did you test it out by mixing a legacy cervice upstream and downstream and see if it is still working as expected?

File: packages/simcore-sdk/src/simcore_sdk/node_ports_common/data_items_utils.py (hunk 1 above)
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_common/data_items_utils.py:20]: not exactly all of it... storage also needed a bit of love to accept this

**Comment by sanderegg** [comment]
> I have one concern. Did you test it out by mixing a legacy cervice upstream and downstream and see if it is still working as expected?

@GitHK : legacy service downstream (jupyter-python-math 1.6.11) works. I cannot guarantee that any legacy service will work since this would need many tests and they are supposed to become deprecated anyway.

**Comment by GitHK** [comment]
> > I have one concern. Did you test it out by mixing a legacy cervice upstream and downstream and see if it is still working as expected?
> 
> @GitHK : legacy service downstream (jupyter-python-math 1.6.11) works. I cannot guarantee that any legacy service will work since this would need many tests and they are supposed to become deprecated anyway. Upstream there is no issue.

@sanderegg that's enough for me. I'm fine with the changes at this point. Thanks for checking.

**Review by pcrespov — APPROVED** [review]
👍

**Review by odeimaiz — APPROVED** [review]

**Comment by elisabettai** [comment]
Target-tested by creating a file with the same name in all the output ports (of a jupyter-math v2.0.8), closing and reopening the study,

**Pull request merged**